docs/advanced_topics/icons.md

```diff
@@ -0,0 +1,90 @@
+(icons)=
+
+# Icons
+
+Wagtail comes with an icon set. The icons are used throughout the admin interface.
+
+Elements that use icons are:
+
+- [Register Admin Menu Item](register_admin_menu_item)
+- [Clientside components](extending_clientside_components)
+- [Rich text editor toolbar buttons](extending_the_draftail_editor)
+- [ModelAdmin menu](modeladmin_menu_icon)
+- [Streamfield blocks](custom_streamfield_blocks)
+
+This document describes how to choose and add icons.
+
+## Available icons and their names
+
+Icons are registered in `wagtail/admin/wagtail_hooks.py`. 
+The _filename_ without the extension is the icon name.
+
+Enable the [styleguide](styleguide) to view icons and their names.
+
+[//]: # (The code is included to present an up-to-date icon name list)
+
+```{eval-rst}
+.. literalinclude:: ../../wagtail/admin/wagtail_hooks.py
+   :language: python
+   :pyobject: register_icons
+```
+
+## Add a custom icon
+
+Draw or download an icon.
+
+The `svg` tag should:
+
+- Set `id="icon-<name>"` attribute, icons are referenced by this name 
+- Set `xmlns="http://www.w3.org/2000/svg"` attribute
+- Set `viewBox="..."` attribute
```

Comment:
```suggestion
- Set the `id="icon-<name>"` attribute. Icons are referenced by this name 
- Set the `xmlns="http://www.w3.org/2000/svg"` attribute
- Set the `viewBox="..."` attribute
```

Reply:
Is there any guidance on the contents of `viewBox`? I have done several of these icons so far, and every time have to just make up guesses as to what numbers to put here until it eventually looks OK for that specific icon. It's an extremely frustrating developer experience. Many existing SVGs do not have this field, so the developer will likely have to guess.

Reply:
@vsalvino A tool like https://jakearchibald.github.io/svgomg/ has the option `Prefer viewBox to width/height`. 

AFAIK the viewbox attribute works better when scaling the icon to the target size.